scripts/test-certificates.sh

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bash
+set -ueo pipefail
+source $(dirname $0)/utils.sh
+
+TMPDIR=$(mktemp -d -t tmp.XXXXXXXXXX)
+IDENTS_DIR=$TMPDIR/identities
+CERTS_DIR=$TMPDIR/certificates
+# TODO: make port numbers more dynamic
+CERTS_ADDR=127.0.0.1:11000
+CERTS_ADDR_PRIV=127.0.0.1:11001
+
+# TODO: find a better way
+kill_certificates_server() {
+  killall "certificates"
+}
+
+cleanup() {
+  if [[ -n $(ps | grep "certificates") ]]; then
+    kill_certificates_server
+  fi
+  rm -rf "$TMPDIR"
+  echo "cleaned up test successfully"
+}
+
+trap cleanup EXIT INT
+
+_certificates() {
+  subcommand=$1
+  shift
+
+  ident_dir="${IDENTS_DIR}/certificates"
+  ca_cert_path="${ident_dir}/ca.cert"
+  ca_key_path="${ident_dir}/ca.key"
+  rev_dburl="bolt://${CERTS_DIR}/revocations.db"
+
+  # NB: `--identity-dir` and `--config-dir` flags are only bound globally to subcommands
+  certificates --identity-dir "$ident_dir" \
+               --config-dir "$CERTS_DIR" \
+               "$subcommand" \
+               --signer.ca.cert-path "$ca_cert_path" \
+               --signer.ca.key-path "$ca_key_path" \
+               --server.address "$CERTS_ADDR" \
+               --server.private-address "$CERTS_ADDR_PRIV" \
+               --server.revocation-dburl="$rev_dburl" \
+               --log.level warn \
+                "$@"
+}
+
+_identity() {
+  subcommand=$1
+  rev_dburl="bolt://${IDENTS_DIR}/revocations.db"
+  shift
+
+  # NB: `--identity-dir` and `--config-dir` flags are only bound globally to subcommands
+  identity --identity-dir "$IDENTS_DIR" \
+           "$subcommand" \
+           --signer.tls.revocation-dburl "$rev_dburl" \
+           --log.level info \
+           "$@"
+}
+
+_identity_create() {
+  _identity create  $1 --difficulty 0 --concurrency 1 >/dev/null
+}
+
+_identity_create 'certificates'
+_certificates setup
+
+for i in {0..4}; do
+  email="testuser${i}@mail.example"
+  ident_name="testidentity${i}"
+
+  _identity_create $ident_name
+
+  if [[ i -gt 0 ]]; then
+    _certificates auth create "$i" "$email"
+  fi
+done
+
+exported_auths=$(_certificates auth export)
+_certificates run --signer.min-difficulty 0 &
+
+sleep 1
```

Comment:
Would this cause flakiness, if things don't complete in 1s?

Reply:
Yeah it should..

Reply:
This is just waiting for the certificates server to come up.